server/queues/tasks/ImportMarkdownZipTask.ts

```diff
@@ -0,0 +1,215 @@
+import path from "path";
+import JSZip from "jszip";
+import { find } from "lodash";
+import mime from "mime-types";
+import { v4 as uuidv4 } from "uuid";
+import documentImporter from "@server/commands/documentImporter";
+import Logger from "@server/logging/logger";
+import { FileOperation, User } from "@server/models";
+import ImportTask, { StructuredImportData } from "./ImportTask";
+
+type FileTreeNode = {
+  /** The title, extracted from the file name */
+  title: string;
+  /** The file name including extension */
+  name: string;
+  /** The full path to within the zip file */
+  path: string;
+  /** The nested children */
+  children: FileTreeNode[];
+};
+
+export default class ImportMarkdownZipTask extends ImportTask {
+  public async parseData(
```

Comment:
`parseData` is now the only method that custom importers have to implement it. It accepts the file as input and `StructuredImportData` object as output which will be persisted by the parent class.